src/v/cloud_storage/manifest.cc

```diff
@@ -0,0 +1,442 @@
+    Writer<OStreamWrapper> w(wrapper);
+    w.StartObject();
+    w.Key("version");
+    w.Int(static_cast<int>(manifest_version::v1));
+    w.Key("namespace");
+    w.String(_topic_config->tp_ns.ns());
+    w.Key("topic");
+    w.String(_topic_config->tp_ns.tp());
+    w.Key("partition_count");
+    w.Int(_topic_config->partition_count);
+    w.Key("replication_factor");
+    w.Int(_topic_config->replication_factor);
+    w.Key("revision_id");
+    w.Int(_rev());
+
+    // optional values are encoded in the following manner:
+    // - key set to null - optional is nullopt
+    // - key is not null - optional has value
+    w.Key("compression");
+    if (_topic_config->properties.compression.has_value()) {
+        w.String(boost::lexical_cast<std::string>(
+          *_topic_config->properties.compression));
+    } else {
+        w.Null();
+    }
+    w.Key("cleanup_policy_bitflags");
+    if (_topic_config->properties.cleanup_policy_bitflags.has_value()) {
+        w.String(boost::lexical_cast<std::string>(
+          *_topic_config->properties.cleanup_policy_bitflags));
+    } else {
+        w.Null();
+    }
+    w.Key("compaction_strategy");
+    if (_topic_config->properties.compaction_strategy.has_value()) {
+        w.String(boost::lexical_cast<std::string>(
+          *_topic_config->properties.compaction_strategy));
+    } else {
+        w.Null();
+    }
+    w.Key("timestamp_type");
+    if (_topic_config->properties.timestamp_type.has_value()) {
+        w.String(boost::lexical_cast<std::string>(
+          *_topic_config->properties.timestamp_type));
+    } else {
+        w.Null();
+    }
+    w.Key("segment_size");
+    if (_topic_config->properties.segment_size.has_value()) {
+        w.Int64(*_topic_config->properties.segment_size);
+    } else {
+        w.Null();
+    }
+    // NOTE: manifest_object_name is intentionaly ommitted
+
+    // tristate values are encoded in the following manner:
+    // - key not present - tristate is disabled
+    // - key set to null - tristate is enabled but not set
+    // - key is not null - tristate is enabled and set
+    if (!_topic_config->properties.retention_bytes.is_disabled()) {
+        w.Key("retention_bytes");
+        if (_topic_config->properties.retention_bytes.has_value()) {
+            w.Int64(_topic_config->properties.retention_bytes.value());
+        } else {
+            w.Null();
+        }
+    }
+    if (!_topic_config->properties.retention_duration.is_disabled()) {
+        w.Key("retention_duration");
+        if (_topic_config->properties.retention_duration.has_value()) {
+            w.Int64(
+              _topic_config->properties.retention_duration.value().count());
+        } else {
+            w.Null();
+        }
+    }
+    w.EndObject();
+}
+
+remote_manifest_path topic_manifest::get_manifest_path() const {
+    // The path is <prefix>/meta/<ns>/<topic>/topic_manifest.json
+    vassert(_topic_config, "Topic config is not set");
```

Comment:
can you print `*this` in the `vassert()` to make sure we get as much information about the core dump as we can